### ISSUE TITLE
peering request 'QuerySet' object has no attribute 'peeringdb_network'

### ISSUE THREAD

**Issue by freedomwarrior** [peering-manager/peering-manager#478]
`Python 3.8.10`
`peering-manager (v1.4.5-dev)`

Email template form doc.

When trying to generate some peering request email, got error:
`'QuerySet' object has no attribute 'peeringdb_network'`

**Comment by jamesditrapani** [comment]
@freedomwarrior Can you provide the template you're using?

**Comment by freedomwarrior** [comment]
> @freedomwarrior Can you provide the template you're using?

```
{%- set local_as = affiliated_autonomous_systems | get(asn=64500) %}
Dear {{ autonomous_system.name }},

We are AS {{ local_as.asn }} peering team,

We found that we could peer at the following IXPs:
{%- for ixp in autonomous_system | shared_ixps(local_as) %}
  - IX Name: {{ ixp.name }}
    AS{{ local_as.asn }} details:
      IPv6: {{ ixp | local_ips(6) | join(", ") }}
      IPv4: {{ ixp | local_ips(4) | join(", ") }}

    AS{{ autonomous_system.asn }} details:
    {%- for s in autonomous_system | missing_sessions(local_as, ixp) %}
      * {{ s.ipaddr4 }}
      * {{ s.ipaddr6 }}
    {%- endfor %}
{%- endfor %}

If you want to peer with us, you can configure the sessions and reply to this
email. We will configure them as well if you agree to do so.

Kind regards,

--
Your awwesome peering team
```

**Comment by jamesditrapani** [comment]
@freedomwarrior Thanks. I will try and reproduce this. Do you have the full error/traceback you got?

**Comment by rarsenault** [comment]
@freedomwarrior Change the "64500" in "get(asn=64500)" to your preferred affiliated ASN. 

Noticed the local_ips(x) doesn't work though, no matter what you pass to it, it returns all IPv4/IPv6 peering addresses, ex:

![image](https://user-images.githubusercontent.com/46567135/129822818-68e3288a-99b4-425a-94ef-3cc70e016a76.png)

@gmazoyer apologies for any protocol violations, I'm not much of a dev or a github user, but locally i got the expected behaviour with the following in jinja2.py, i'm confident someone can come up with a more aesthetically pleasing way:

```
def local_ips(value, family=0):
    """
    Returns local IP addresses for a BGP session or an IXP.
    """
    if isinstance(value, DirectPeeringSession):
        return value.local_ip_address

    if isinstance(value, InternetExchangePeeringSession):
        if value.ip_address.version == 6:
            return value.ixp_connection.ipv6_address
        else:
            return value.ixp_connection.ipv4_address

    if isinstance(value, InternetExchange):
        ips = []
        for c in Connection.objects.filter(internet_exchange_point=value):
            if c.ipv4_address:
                ips.append(c.ipv4_address)
            if c.ipv6_address:
                ips.append(c.ipv6_address)
        if family not in (4, 6):
          return ips
        else:
          ips = [ip for ip in ips if ip.version == family]
          return ips

    return None
```

**Comment by gmazoyer** [comment]
> Noticed the local_ips(x) doesn't work though, no matter what you pass to it, it returns all IPv4/IPv6 peering addresses, ex:
> 
> @gmazoyer apologies for any protocol violations, I'm not much of a dev or a github user, but locally i got the expected behaviour with the following in jinja2.py, i'm confident someone can come up with a more aesthetically pleasing way:

Fixed in 944a2b7

**Comment by freedomwarrior** [comment]
> @freedomwarrior Change the "64500" in "get(asn=64500)" to your preferred affiliated ASN.
> 
> Noticed the local_ips(x) doesn't work though, no matter what you pass to it, it returns all IPv4/IPv6 peering addresses, ex:
> 
> ![image](https://user-images.githubusercontent.com/46567135/129822818-68e3288a-99b4-425a-94ef-3cc70e016a76.png)
> 
> @gmazoyer apologies for any protocol violations, I'm not much of a dev or a github user, but locally i got the expected behaviour with the following in jinja2.py, i'm confident someone can come up with a more aesthetically pleasing way:
> 
> ```
> def local_ips(value, family=0):
>     """
>     Returns local IP addresses for a BGP session or an IXP.
>     """
>     if isinstance(value, DirectPeeringSession):
>         return value.local_ip_address
> 
>     if isinstance(value, InternetExchangePeeringSession):
>         if value.ip_address.version == 6:
>             return value.ixp_connection.ipv6_address
>         else:
>             return value.ixp_connection.ipv4_address
> 
>     if isinstance(value, InternetExchange):
>         ips = []
>         for c in Connection.objects.filter(internet_exchange_point=value):
>             if c.ipv4_address:
>                 ips.append(c.ipv4_address)
>             if c.ipv6_address:
>                 ips.append(c.ipv6_address)
>         if family not in (4, 6):
>           return ips
>         else:
>           ips = [ip for ip in ips if ip.version == family]
>           return ips
> 
>     return None
> ```

Thanks, all workes good!
